lib/safe_redirect.rb

```diff
@@ -0,0 +1,17 @@
+# -*- encoding : utf-8 -*-
+#
+# Public: Sanitise a redirect parameter
+require 'uri'
+
+class SafeRedirect
+  attr_reader :redirect_parameter
+
+  def initialize(redirect_parameter)
+    @redirect_parameter = redirect_parameter
+  end
+
+  def path
+    uri = URI.parse(redirect_parameter)
+    URI::Generic.build(path: uri.path, fragment: uri.fragment).to_s
```

Comment:
`redirect_to` can take a URI object to calling `#to_s` is only needed here because we are appending the `emergency` query param in one place. Could think about passing in query params as part of the `initialize` method to be used when building the URI.